internal/contributebot/process/process.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package process maps Github events to actions, and implements the actions.
+package process
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/google/go-github/github"
+)
+
+const (
+	inProgressLabel = "in progress"
+)
+
+// IssueData is information about an issue event.
+// See the github documentation for more details about the fields:
+// https://godoc.org/github.com/google/go-github/github#IssuesEvent
+type IssueData struct {
+	// Action that this event is for.
+	// Possible values are: "assigned", "unassigned", "labeled", "unlabeled", "opened", "closed", "reopened", "edited".
+	Action string
+	// Issue the event is for.
+	Issue *github.Issue
+	// Change made as part of the event.
+	Change *github.EditChange
+}
+
+func (i *IssueData) String() string {
+	return fmt.Sprintf("[%s issue #%d]", i.Action, i.Issue.GetNumber())
+}
+
+// Action represents an action to be taken.
+type Action interface {
```

Comment:
Rather than an interface, if you use a struct to describe the result of processing an issue, you get a more testable result. Something like:

```go
type IssueEditList struct {
  Close bool
  AddLabels []string
  RemoveLabels []string
  AddComment string
}
```

(I realize "edit list" is probably not the most precise term here: I'm borrowing it from the world of film editing where it is the precise term for it.)